src/api/index.js

```diff
@@ -0,0 +1,272 @@
+            data: {
+                method: 'GET'
+            },
+        })
+    };
+
+    fetchSongs = (page, limit) => {
+        return this.makeRequest(
+            {
+                url: `${this.url}/songs?page=${page}&limit=${limit}`,
+                data: {
+                    method: 'GET'
+                },
+                token: localStorage.token
+
+            })
+    };
+
+    fetchLikeSong = id => {
+        return this.makeRequest({
+            url: `${this.url}/songs/${id}/like`,
+            data: {
+                method: 'POST',
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchFindSong = (title, artist) => {
+        return this.makeRequest({
+            url: `${this.url}/songs/search?title=${title}&artist=${artist}`,
+            data: {
+                method: 'GET',
+
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchLikedSongs = () => {
+        return this.makeRequest({
+            url: `${this.url}/songs/liked`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchSongsByArtist = (artist) => {
+        return this.makeRequest({
+            url: `${this.url}/songs?artist=${artist}`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchTranslateWord = (word) => {
+        return this.makeRequest({
+            url: `${this.url}/words/translate?word=${word}`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchDataUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/user`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchUserUpdateData = (data) => {
+        return this.makeRequest({
+            url: `${this.url}/user`,
+            data: {
+                method: 'PATCH',
+                body: JSON.stringify(data)
+            },
+            token: localStorage.token
+        })
+    };
+    fetchSetPassword = (user) => {
+        return this.makeRequest({
+            url: `${this.url}/setPassword`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    password: user.password,
+                })
+            },
+            token: localStorage.token
+        })
+    };
+    fetchResetPassword = (user) => {
+        return this.makeRequest({
+            url: `${this.url}/resetPassword`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    password: user.password,
+                    old_password: user.old_password,
+                })
+            },
+            token: localStorage.token
+        })
+    };
+}
+
+
+const Api = new ApiClass();
```

Comment:
```suggestion
export default new ApiClass();
```
I think this should also work properly